http/req/validate.go

```diff
@@ -0,0 +1,110 @@
+package req
+
+import (
+	"errors"
+	"reflect"
+	"strings"
+
+	v10 "github.com/go-playground/validator/v10"
+	"github.com/xy-planning-network/trails"
+)
+
+type validator struct {
+	valid *v10.Validate
+}
+
+// newValidator constructs a validator, which applies default configuration.
+func newValidator() validator {
+	v := v10.New()
+	v.RegisterValidation("enum", validateEnumerable)
+	v.RegisterTagNameFunc(func(field reflect.StructField) string {
+		name := strings.SplitN(field.Tag.Get("json"), ",", 2)[0]
+		if name == "-" {
+			name = ""
+		}
+
+		if name == "" {
+			name = strings.SplitN(field.Tag.Get("schema"), ",", 2)[0]
+		}
+
+		if name == "-" {
+			name = ""
+		}
+
+		return name
+	})
+
+	return validator{v}
+}
+
+// validate checks the fields on structPtr match the rules set by "validate" struct tags.
+// On success, validate returns no error.
+// On failure, validate translates each issue to a ValidationError,
+// returning them all as ValidationErrors.
+func (v validator) validate(structPtr any) error {
+	err := v.valid.Struct(structPtr)
```

Comment:
Name-spacing with `valid` adds confusion here. I think that embedding directly `validator.Struct(structPtr)` or `v.Struct(structPtr)` removes the unhelpful `.valid` step call as that appears more like an attribute.